jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -3266,6 +3269,80 @@ ecma_op_ordinary_object_has_own_property (ecma_object_t *object_p, /**< the obje
   return property != ECMA_PROPERTY_TYPE_NOT_FOUND && property != ECMA_PROPERTY_TYPE_NOT_FOUND_AND_STOP;
 } /* ecma_op_ordinary_object_has_own_property */
 
+#if ENABLED (JERRY_ESNEXT)
+
+/**
+ * Set a weak reference from a container to a key object
+ */
+void
+ecma_op_object_set_weak (ecma_object_t *const object_p, /**< key object */
+                         ecma_extended_object_t *const target_p, /**< target object */
```

Comment:
replace `ecma_extended_object_t ` with `ecma_object_t ` sincse it's not a requirement.